go/ct/driver/probe.go

```diff
@@ -0,0 +1,198 @@
+//
+// Copyright (c) 2024 Fantom Foundation
+//
+// Use of this software is governed by the Business Source License included
+// in the LICENSE file and at fantom.foundation/bsl11.
+//
+// Change Date: 2028-4-16
+//
+// On the date above, in accordance with the Business Source License, use
+// of this software will be governed by the GNU Lesser General Public Licence v3
+//
+
+package main
+
+import (
+	"fmt"
+	"runtime"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"pgregory.net/rand"
+
+	"github.com/Fantom-foundation/Tosca/go/ct"
+	"github.com/Fantom-foundation/Tosca/go/ct/common"
+	cliUtils "github.com/Fantom-foundation/Tosca/go/ct/driver/cli"
+	"github.com/Fantom-foundation/Tosca/go/ct/gen"
+	"github.com/Fantom-foundation/Tosca/go/ct/rlz"
+	"github.com/Fantom-foundation/Tosca/go/ct/spc"
+	"github.com/Fantom-foundation/Tosca/go/ct/st"
+	"github.com/dsnet/golib/unitconv"
+	"github.com/urfave/cli/v2"
+	"golang.org/x/exp/maps"
+)
+
+var ProbeCmd = cli.Command{
+	Action:    doProbe,
+	Name:      "probe",
+	Usage:     "Run random tests on an EVM implementation",
+	ArgsUsage: "<EVM>",
+	Flags: []cli.Flag{
+		cliUtils.FilterFlag,
+		cliUtils.SeedFlag,
+		&cli.IntFlag{
+			Name:  "timeout",
+			Usage: "maximum time in minutes to run the probing. Default 30 minutes",
+			Value: 30,
+		},
+	},
+}
+
+func doProbe(context *cli.Context) error {
+	var evmIdentifier string
+	if context.Args().Len() >= 1 {
+		evmIdentifier = context.Args().Get(0)
+	}
+
+	evm, ok := evms[evmIdentifier]
+	if !ok {
+		return fmt.Errorf("invalid EVM identifier, use one of: %v", maps.Keys(evms))
+	}
+
+	jobCount := cliUtils.JobsFlag.Fetch(context)
+	if jobCount <= 0 {
+		jobCount = runtime.NumCPU()
+	}
+
+	seed := cliUtils.SeedFlag.Fetch(context)
+
+	timeout := time.Duration(context.Int("timeout")) * time.Minute
+	hasTimeouted := false
+
+	// The constraints to be placed on generated states.
+	condition := rlz.And(
+		rlz.IsCode(rlz.Pc()),
+		// update to cancun when all necessary changed have been implemented
+		rlz.RevisionBounds(common.MinRevision, common.R12_Shanghai),
```

Comment:
I doubt that we will remember updating this. 

Can we replace `common.R12_Shanghai` with `common.NewestSupportedRevision`?
If not, maybe we can introduce a `common.NewestFullySpecifiedRevision` for these kind of use cases?

Reply:
`common.NewestSupportedRevision` is already Cancun. 

not sure introducing another concept that we also need to eventually update is better than leaving  a `TODO`.

Reply:
I would use newest every time. today we are Cancun, but this will certainly change and then we would have to revisit this code. Your rule is defined for the foreseeable future, that is newest supported.

Reply:
the problem with using `NewestSupportedRevision` is that we don't currently support all instructions in that set, so this command would fail. I'm ok with halting this PR until we have full support for `Cancun`